src/Homepage/index.js

```diff
@@ -367,7 +416,13 @@ class Homepage extends React.Component {
       lng: Number(res.lng),
     }));
     const { translations } = this.props;
-    return (
+    return this.state.showPageLoader ? (
+      <div className="text-center py-5">
+        <div className="spinner-border text-secondary" role="status">
+          <span className="sr-only">Loading...</span>
+        </div>
+      </div>
+    ) : (
```

Comment:
Maybe we can move this component to App level as well i think. And pass the `togglePageLoader` method in global context to all its child components. @arjun27 what do you think ?